test/instrument/test.ml

```diff
@@ -0,0 +1,43 @@
+open Ppxlib
+
+type operation = Int of (int -> int) | Float of (float -> float)
+[%%expect{|
+type operation = Int of (int -> int) | Float of (float -> float)
+|}]
+
+let apply_operation operation = object
+  inherit Ast_traverse.map as super
+
+  method! expression e =
+    match (e.pexp_desc, operation) with
+    | (Pexp_constant (Pconst_integer (x, _)), Int (operation)) -> Ast_builder.Default.eint ~loc:e.pexp_loc @@ operation @@ int_of_string x
+    | (Pexp_constant (Pconst_float (x, _)), Float(operation))-> Ast_builder.Default.efloat ~loc:e.pexp_loc @@ string_of_float @@ operation @@ float_of_string x
+    | _ -> super#expression e
+end
+[%%expect{|
+val apply_operation : operation -> Ast_traverse.map = <fun>
+|}]
+
+let () = 
+  let transform = Int ((+) 1) |> apply_operation in
+  let instrument = Instrument.make ~position:Before ~transformation:transform#structure in
+  Driver.register_transformation ~instrument "a"
+
+let () = 
+  let transform = Float (( +. ) 1.) |> apply_operation in
+  let instrument = Instrument.make ~position:After ~transformation:transform#structure in
+  Driver.register_transformation ~instrument "b"
+
+let () =
+  let transform = Int (abs) |> apply_operation in
+  Driver.register_transformation ~impl:transform#structure "c"
+
+let () =
+  let transform = Float (Float.abs) |> apply_operation in
+  Driver.register_transformation ~impl:transform#structure "d"
+
+let x, y = -1, -1.0
+[%%expect{|
+val x : int = 0
+val y : float = 2.
+|}]
```

Comment:
This test seems good, however for readers to convince themselves that the test is passing they have to carefully look at the various transformations and apply them in their head. It would be better if things where completely obvious. One way to do it would be to extend a list of strings with the transformation name. For instance, the expected result would be:

```ocaml
["instrumentation pos=Before"; "impl"; "instrumentation pos=After"]
```

Then a reader can immediately see whether the test is passing or not.

Reply:
That's a very nice idea, thanks! I've added the "a"-, "b"- and "c"-prefixes to the names to stress that the alphabetic order doesn't influence. But if you find that distracting, I can get rid of them.

Reply:
Looks good to me :)

Reply:
BTW, the point about using the letter prefixes to stress that the alphabetic order doesn't influence the result is the kind of thing that is worth explaining in a comment. The intent is not necessarily obvious when reading the code, and if it's not obvious to a future maintainer they might simplify it away.

Reply:
I've made a comment explaining that.